### PR TITLE
CVSL-442 support user can get licence stats

### PR DESCRIPTION
**Pull request by GurnankCheema**

### CODE REVIEW

File: server/data/licenceApiClient.ts [hunk 1]
```diff
@@ -79,6 +80,17 @@ export default class LicenceApiClient extends RestClient {
...
   }
 
+  async getlicenceStatistics(startDate: string, endDate: string, user: User): Promise<LicenceStatistics[]> {
+    try {
+      return (await this.get(
+        { path: `/support/licence-statistics?startDate=${startDate}&endDate=${endDate}` },
+        { username: user.username }
+      )) as Promise<LicenceStatistics[]>
+    } catch (error) {
+      return error.status >= 400 && error.status < 500 ? null : error
```
Comment (andrewrlee) [inline server/data/licenceApiClient.ts:90]: I don't think there are any expected errors for this endpoint so don't need to try/catch

The other calls wrap as absent OMU is modelled as a 404.

File: server/routes/support/index.ts [hunk 2]
```diff
@@ -11,7 +11,9 @@ import validationMiddleware from '../../middleware/validationMiddleware'
 import prisonIdCurrent from './types/prisonIdCurrent'
 import prisonIdAndEmail from './types/prisonIdAndEmail'
 import prisonIdDelete from './types/prisonIdDelete'
+import dateRange from './types/dateRange'
```
Comment (andrewrlee) [inline server/routes/support/index.ts:14]: This is all a bit weird.
These are references to the classes (types), so should all be upper cased.

File: server/routes/support/types/customValidators.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+function isStartBeforeEnd(startDate: string, endDate: string) {
+  const parsedStartDate = startDate.split('/')
+  const parsedEndDate = endDate.split('/')
+  const StartDateDate = parseInt(parsedStartDate[0], 10)
+  const StartDateMonth = parseInt(parsedStartDate[1], 10)
+  const StartDateYear = parseInt(parsedStartDate[2], 10)
+
+  const EndDateDate = parseInt(parsedEndDate[0], 10)
+  const EndDateMonth = parseInt(parsedEndDate[1], 10)
+  const EndDateYear = parseInt(parsedEndDate[2], 10)
```
Comment (andrewrlee) [inline server/routes/support/types/customValidators.ts:53]: Is there a utility method in date-fns to parse the date?  

Will this always be called or only if `IsDateFormat` is true?

File: server/services/licenceService.ts [hunk 4]
```diff
@@ -151,6 +151,11 @@ export default class LicenceService {
     return this.licenceApiClient.createLicence(licence, user)
   }
 
+  async getlicenceStatistics(startDate: string, endDate: string, user: User) {
+    const data = await this.licenceApiClient.getlicenceStatistics(startDate, endDate, user)
```
Comment (andrewrlee) [inline server/services/licenceService.ts:155]: Are you converting from `d/mm/yyyy` -> `yyyy-mm-dd` somewhere?

Comment (GurnankCheema) [reply server/services/licenceService.ts:155]: yes

File: server/data/licenceApiClient.ts (hunk 1 above)
Comment (GurnankCheema) [reply server/data/licenceApiClient.ts:90]: ok

File: server/routes/support/index.ts (hunk 2 above)
Comment (GurnankCheema) [reply server/routes/support/index.ts:14]: ok

File: server/routes/support/types/customValidators.ts (hunk 3 above)
Comment (GurnankCheema) [reply server/routes/support/types/customValidators.ts:53]: I think I need to revisit this

**Pull request closed without merging**